01/app.js

```diff
@@ -1,15 +1,16 @@
 function getProductList() {
-    return [
-        {name: 'JavaScript: podstawy', type: 'book', count: 3, price: 67.19, discount: 0.21},
-        {name: 'React: podstawy', type: 'book', count: 4, price: 79.17, discount: 0.27}
-    ]
+	return [
+		{ name: 'JavaScript: podstawy', type: 'book', count: 3, price: 67.19, discount: 0.21 },
+		{ name: 'React: podstawy', type: 'book', count: 4, price: 79.17, discount: 0.27 },
+	];
 }
 
 function getTotalPrice(productList) {
-    return productList.reduce( (product, acc) => {
-        return acc + (product.price * product.count);
-    });
+	return productList.reduce((acc, product) => {
+		return acc + product.price * product.count * (1 - product.discount);
+	}, 0);
 }
 
-const totalPrice = getTotalPrice( getProductList() );
-console.log(totalPrice); // prawidłowa wartość: 390.42 (należy zaaokrąglić do 2 miejsc po przecinku)
\ No newline at end of file
+const totalPrice = getTotalPrice(getProductList());
+console.log(totalPrice.toFixed(2));
+//debugger nic mi nie powiedzial, ale gpt odrazu wyczul o co chodzi... tylko sam musialem dodac discount...
```

Comment:
Postaraj się teraz (jak już wiesz gdzie jest błąd) odpalić debugger i zrozumieć co się dzieje, aby następnym razem było łatwiej z debuggerem :)

Reply:
w tym przykladzie debbuger nic mi kompletnie nie mowi, bo pokazuje ze acc = [obiekt] i pokaze tylko jedno dodanie kiedy to acc = sumie pierwszego zamowienia i to jest koniec. Bez debuggera w tym przykladzie samemu rozpoznaje co sie w kodzie dzieje. co nie zmienia faktu ze acc z productem nie wzbudzalo u mnie podejrzen, ze kolejność byla podmienona